rfcs/rfc-colin-states-mapping-to-lear.md

```diff
@@ -0,0 +1,154 @@
+            'restoration': { # target filing
+                # invalid target filing if filing that transitioned business to its current state is in list
+                'fullRestoration': ['registration', 'incorporationApplication'], 
+                'limitedRestoration': ['registration', 'incorporationApplication'], 
+            }
+            ...
+        },
+        Business.State.HISTORICAL: {
+            'restoration': { 
+                'fullRestoration': ['almagmation', 'continuationOut', 'continuationIn'], 
+                'limitedRestoration': ['almagmation', 'continuationOut', 'continuationIn'],
+            },
+            'putBackOn': ['almagmation', 'withdrawal']
+            ...
+        },
+        Business.State.LIQUIDATION: {
+            'restoration': {
+                'fullRestoration': ['liquidation'], 
+                'limitedRestoration': ['liquidation'], 
+            },
+            ...
+        }
+    },
+    'client': {
+        Business.State.ACTIVE: { # current business state
+            ...
+        },
+        Business.State.HISTORICAL: {
+            ...
+        },
+        Business.State.LIQUIDATION: {
+            'restoration': {
+                'fullRestoration': ['liquidation'], 
+                'limitedRestoration': ['liquidation'], 
+            },
+            'putBackOn': ['liquidation']
+            ...
+        }
+    }        
+    
+}
+```
+
+_Allowable state transitions matrix in COLIN for restoration and put back on filings_
+![colin_allowable_filings.png](rfc-colin-states-mapping-to-lear/colin_allowable_filings.png)
+
+Filing validation logic will be updated to include logic to ensure that only valid business state transitions are allowed. 
+
+All filings resulting in business state transitions will need to populate the businesses.state_filing_id field.  This is used to determine if future business state transitions are permitted.
+
+Filings.meta_data field for any filings where a business state transitions occurs will include from/to state.
+
+API consumers such as the FE with a need to determine which filing transitioned the business to its current state will retrieve the business state filing provided by the business response of the GET business endpoint.  For example, this can be used to determine if an ACTIVE business is currently in _limited restoration_.
+
+# Adoption strategy
+
+The _LEAR_ team will create tickets to implement the required components.
+
+# Unresolved questions
+
+There are still some details that need to be fleshed out for filings like the ones below. These details will be worked out as we implement the filings in LEAR.
+
+**Involuntary dissolution** - this is currently a manual process that transitions a business to D1 to D2 based on some conditions. Uncertain as to how this changes in LEAR and how we will represent the involuntary dissolution process from a data and process perspective. Will the manual process become an automated process in LEAR?
```

Comment:
We will need to know if we furnishing a customer with their dissolution status is necessary. D1 & D2 are time sentinels that raise the warning level as a business moves to being dissolved for failure to file.

Reply:
PI plan for entities has a sprint for figuring digging into the details of involuntary dissolution.  Will make sure to ask the business regarding your comment about whether we need to furnish customer with their dissolution status.   

Also added a note about this in the unresolved questions section.